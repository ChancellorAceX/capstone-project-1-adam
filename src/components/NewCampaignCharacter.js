import React from 'react';

export default function NewCampaignCharacter(props) {
  const { character, monstername, pcclass, level, ac, maxhp } = props.character;

  return (
    <tr>
      <th>{!character ? monstername + ' ' + (props.index > 0 ? props.index : '') : character}</th>
      <th>{!pcclass ? monstername === 'Name' ? 'Type' : monstername : `${pcclass}: ${level}`}</th>
      <th>{ac}</th>
      <th>{maxhp}</th>
    </tr>
  );
}