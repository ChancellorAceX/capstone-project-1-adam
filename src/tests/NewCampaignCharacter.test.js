import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NewCampaignCharacter from '../components/NewCampaignCharacter';

describe('NewCampaignCharacter component', () => {
  const props = {
    "character": {
      "character": "name",
      "pcclass": "class",
      "ac": "10",
      "level": "1",
      "maxhp": "14"
    }
  };
  it('renders the full NewCampaignCharacter component', () => {
    const wrapper = shallow(<NewCampaignCharacter {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});