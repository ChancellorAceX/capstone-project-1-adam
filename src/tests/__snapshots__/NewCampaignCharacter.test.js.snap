// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCampaignCharacter component renders the full NewCampaignCharacter component 1`] = `
<tr>
  <th>
    name
  </th>
  <th>
    class: 1
  </th>
  <th>
    10
  </th>
  <th>
    14
  </th>
</tr>
`;
